### PR TITLE
Add troubleshooting docs for some Windows issues

### PR DESCRIPTION
**Pull request by klahnakoski**
This are the changes required for windows install, and running of tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,3 +58,23 @@ and are excluded from the default test run. But if needed, you can run them loca
...
+    poetry export --dev --without-hashes -f requirements.txt > requirements.in
+    pip-compile --upgrade --generate-hashes --output-file requirements.txt requirements.in 
+    pip install -r requirements.txt
+
+### Running tests
+
+Tests are run from the main `mozci` project directory. Be sure you have activated the virtual environment before running tests (`.venv\Scripts\activate`)
+
+    set PYTHONPATH=.
+    pytest tests
```
Comment (klahnakoski) [inline README.md]: I have not got `tox` working yet

File: README.md [hunk 2]
```diff
@@ -58,3 +58,23 @@ and are excluded from the default test run. But if needed, you can run them loca
...
 ```
+
+### Windows installation
+
+The `poetry install` command may lock up on Windows10 (Python3.7.6) you can get around this with:
+
+    python -m pip install virtualenv
+    python -m virtualenv .venv                   # IMPORTANT: Notice the dot in the name
+    .venv\Scripts\activate
+    pip install pip-tools
```
Comment (klahnakoski) [inline README.md]: `pip-tools` is for `pip-compile`

File: README.md [hunk 3]
```diff
@@ -58,3 +58,23 @@ and are excluded from the default test run. But if needed, you can run them loca
...
+    poetry export --dev --without-hashes -f requirements.txt > requirements.in
+    pip-compile --upgrade --generate-hashes --output-file requirements.txt requirements.in
```
Comment (klahnakoski) [inline README.md]: The poetry hashes (if they are exported) are incomplete, so we use `pip-compile` to list all transitive dependencies.

File: README.md [hunk 4]
```diff
@@ -58,3 +58,23 @@ and are excluded from the default test run. But if needed, you can run them loca
...
+
+The `poetry install` command may lock up on Windows10 (Python3.7.6) you can get around this with:
+
+    python -m pip install virtualenv
+    python -m virtualenv .venv                   # IMPORTANT: Notice the dot in the name
+    .venv\Scripts\activate
+    pip install pip-tools
+    poetry export --dev --without-hashes -f requirements.txt > requirements.in
+    pip-compile --upgrade --generate-hashes --output-file requirements.txt requirements.in 
+    pip install -r requirements.txt
```
Comment (klahnakoski) [inline README.md]: once all the requirements are in a file, we can install them

**Comment by armenzg** [comment]
Is there a chance to document the steps to reproduce it and file it upstream?
I've got it working on my Windows VM and without steps to reproduce no one can fix the issue.

**Comment by klahnakoski** [comment]
@armenzg Here are the instruction to reproduce:

    poetry install

This instruction locks up and never completes. If this command works in your environment, then you are using the wrong environment to capture the problem.

The easiest solution is to wait a few years for the project to be mature.  The alternative is to become a `poetry` developer and solve the problem upstream, and hopefully this is not a Heisenbug. I do not like `poetry` enough to do that.

Here is the full run


```
C:\Users\kyle\code-ss\mozci>poetry install -vvv
Creating virtualenv mozci-whf49Q9--py3.7 in C:\Users\kyle\AppData\Local\pypoetry\Cache\virtualenvs
Installing dependencies from lock file


Package operations: 58 installs, 0 updates, 0 removals

  - Installing zipp (3.1.0)
  - Installing appdirs (1.4.3)
  - Installing certifi (2019.11.28)
  - Installing chardet (3.0.4)
  - Installing colorama (0.4.3)
  - Installing distlib (0.3.0)
  - Installing filelock (3.0.12)
  - Installing idna (2.9)
  - Installing importlib-metadata (1.5.0)
  - Installing markupsafe (1.1.1)
  - Installing pyparsing (2.4.6)
  - Installing pytz (2019.3)
  - Installing six (1.14.0)
  - Installing urllib3 (1.25.8)
  - Installing win32-setctime (1.0.1)
  - Installing alabaster (0.7.12)
  - Installing atomicwrites (1.3.0)
  - Installing attrs (19.3.0)
  - Installing babel (2.8.0)
  - Installing cachy (0.3.0)
  - Installing cfgv (3.1.0)
  - Installing docutils (0.16)
  - Installing identify (1.4.13)
  - Installing imagesize (1.2.0)
  - Installing jinja2 (2.11.1)
  - Installing json-e (3.0.2)
  - Installing loguru (0.4.1)
  - Installing markdown (3.2.1)
  - Installing more-itertools (8.2.0)
  - Installing nodeenv (1.3.5)
  - Installing packaging (20.3)

```

at which point it is frozen

**Comment by klahnakoski** [comment]
https://github.com/python-poetry/poetry/issues/2244

**Comment by armenzg** [comment]
Thank you so much for filing upstream!

**Pull request merged**